typescript/sdk/src/consts/chainConnectionConfigs.ts

```diff
@@ -3,206 +3,40 @@ import { ethers } from 'ethers';
-  apiPrefix: 'api-',
-};
-
-export const optimismgoerli: IChainConnection = {
-  provider: new ethers.providers.JsonRpcProvider(
-    'https://goerli.optimism.io',
-    420,
-  ),
-  confirmations: 1,
-  blockExplorerUrl: 'https://goerli-optimism.etherscan.io/',
-  apiPrefix: 'api-',
-};
-
-export const arbitrumgoerli: IChainConnection = {
-  provider: new ethers.providers.JsonRpcProvider(
-    'https://goerli-rollup.arbitrum.io/rpc	',
-    421613,
-  ),
-  confirmations: 1,
-  blockExplorerUrl: 'https://goerli.arbiscan.io',
-  apiPrefix: 'api-',
-};
-
-export const mumbai: IChainConnection = {
-  provider: new ethers.providers.JsonRpcProvider(
-    'https://rpc-mumbai.maticvigil.com',
-    80001,
-  ),
-  confirmations: 30,
-  blockExplorerUrl: 'https://mumbai.polygonscan.com',
-  apiPrefix: 'api-',
-};
-
-export const bsctestnet: IChainConnection = {
-  provider: new ethers.providers.JsonRpcProvider(
-    'https://data-seed-prebsc-1-s3.binance.org:8545',
-    97,
-  ),
-  confirmations: 1,
-  blockExplorerUrl: 'https://testnet.bscscan.com',
-  apiPrefix: 'api-',
-};
-
-export const moonbasealpha: IChainConnection = {
-  provider: new ethers.providers.JsonRpcProvider(
-    'https://rpc.api.moonbase.moonbeam.network',
-    1287,
-  ),
-  confirmations: 1,
-  blockExplorerUrl: 'https://moonbase.moonscan.io',
-  apiPrefix: 'api-',
-};
-
-export const moonbeam: IChainConnection = {
-  provider: new ethers.providers.JsonRpcProvider(
-    'https://rpc.api.moonbeam.network',
-    1284,
-  ),
-  confirmations: 1,
-  blockExplorerUrl: 'https://moonscan.io',
-  apiPrefix: 'api-moonbeam.',
-};
-
-export const test1: IChainConnection = {
-  provider: new ethers.providers.JsonRpcProvider(
-    'http://localhost:8545',
-    31337,
-  ),
-  confirmations: 1,
-};
-
-export const test2: IChainConnection = {
-  provider: new ethers.providers.JsonRpcProvider(
-    'http://localhost:8545',
-    31337,
-  ),
-  confirmations: 1,
-};
-
-export const test3: IChainConnection = {
-  provider: new ethers.providers.JsonRpcProvider(
-    'http://localhost:8545',
-    31337,
-  ),
-  confirmations: 1,
-};
-
-export const chainConnectionConfigs: ChainMap<ChainName, IChainConnection> = {
-  arbitrum,
-  bsc,
-  ethereum,
-  celo,
-  polygon,
-  avalanche,
-  alfajores,
-  fuji,
-  goerli,
-  mumbai,
-  bsctestnet,
-  optimism,
-  moonbasealpha,
-  moonbeam,
-  optimismgoerli,
-  arbitrumgoerli,
-  test1,
-  test2,
-  test3,
-};
+import { objMap } from '../utils/objects';
+
+import { chainMetadata } from './chainMetadata';
+import { Chains } from './chains';
+
+function testChainConnection() {
+  return {
+    provider: new ethers.providers.JsonRpcProvider(
+      'http://localhost:8545',
+      31337,
+    ),
+    confirmations: 1,
+  };
+}
+
+export const chainConnectionConfigs: ChainMap<ChainName, IChainConnection> =
+  objMap(chainMetadata, (chainName, metadata) => {
+    if (chainName.includes('test')) return testChainConnection();
```

Comment:
I think this condition may hit for chains like `bsctestnet`. Maybe a more strict condition is needed to ensure only test1/test2/test3 match

Reply:
Oh great catch, thank you!